enhancements/cluster-logging/workload-identity-auth-for-azure-monitor-logs.md

```diff
@@ -0,0 +1,295 @@
+
+Output
+
+```Go
+// New output type of AzureLogIngestion which can live side by side with the AzureMonitor output
+type AzureLogIngestion struct {
+  // The Data Collection Endpoint (DCE) or Data Collection Rule (DCR) logs ingestion endpoint URL.
+  //
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/logs/logs-ingestion-api-overview#endpoint
+  //
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:XValidation:rule="isURL(self)", message="invalid URL"
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Url",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  URL string `json:"url"`
+
+  // The Data Collection Rule's (DCR) Immutable ID
+  //
+  // A unique identifier for the data collection rule. This property and its value are automatically created when the DCR is created.
+  //
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/data-collection/data-collection-rule-structure#properties
+  //
+  // +kubebuilder:validation:Required
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="DCR Immutable ID",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  DcrImmutableId string `json:"dcrImmutableId"`
+
+  // The stream in the Data Collection Rule (DCR) that should handle the custom data
+  // 
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/data-collection/data-collection-rule-structure#input-streams
+  //
+  // +kubebuilder:validation:Required
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Stream Name",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  StreamName string `json:"streamName"`
+
+  // Authentication sets credentials for authenticating the requests.
+  //
+  // +kubebuilder:validation:Required
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Authentication Options",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  Authentication *AzureLogIngestionAuth `json:"authentication"`
+
+  // Tuning specs tuning for the output
+  //
+  // +kubebuilder:validation:Optional
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Tuning Options"
+  Tuning *BaseOutputTuningSpec `json:"tuning,omitempty"`
+}
+```
+
+Authentication
+
+```Go
+type AzureLogIngestionAuth struct {
+  // ClientId points to the secret containing the client ID used for authentication.
+  // 
+  // This is the application ID that's assigned to your app. You can find this information in the portal where you registered your app.
+  //
+  // +kubebuilder:validation:Required
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Secret with Client ID",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  ClientId *SecretReference `json:"clientId"`
+
+  // TenantId points to the secret containing the tenant ID used for authentication.
+  //
+  // The directory tenant the application plans to operate against, in GUID or domain-name format.
```

Comment:
Might be a place where we can add validation?

Reply:
What kind of validation should we add here? This is information obtained in Azure

Reply:
> GUID or domain-name format

I think we can validate these, though if we dont have precedence then I think it is reasonable to skip.  If you are not getting your logs maybe you need to look closer at auth

Reply:
I think this is more like, did you copy/paste the right information. It's like putting in a password, if you type it wrong it won't be valid. If the GUID or domain-name is wrong, it doesn't matter if it conforms to the right pattern, it is still wrong.

Reply:
at least we can add validation for non-whitespace string
`// +kubebuilder:validation:Pattern=`\S``

Reply:
Does that matter if it's a required field?